### ISSUE TITLE
Feature - Add anti-aliasing option to IPattern

### ISSUE THREAD

**Issue by Youlean** [iPlug2/iPlug2#506]
I have added this with this commit: https://github.com/Youlean/iPlug2-Youlean/commit/bb2f3f8830f588fa3ebd81b301fba106e01e36e5

**Comment by AlexHarker** [comment]
Is there any reason we'd want to leave the option to turn it off?

**Comment by AlexHarker** [comment]
just so I understand correctly, right now it looks like patterns are anti-aliased and this only allows you to switch that off, or have I misunderstood? Thanks.

**Comment by Youlean** [comment]
Yes, this will allow us to turn off AA.

There might be some cases where that is really important like if you want to fill paths one next to another. If you try to place triangles one next to another with sides touching, you can only have a continuous shape without a line in between with AA off.

**Comment by AlexHarker** [comment]
Surely if you want a contours fill you short use a multi-line path? I'm just not clear on a use case which is really necessary.

**Comment by Youlean** [comment]
You are right I could use multi-line path (or two triangles paths), however, that won't be possible if I want to fill 2 paths with different colors.

**Comment by AlexHarker** [comment]
Perhaps, but neither will the sides of your shapes be nicely anti-aliased in that scenario. I'd be interested to see a clear scenario (possibly with pics) where having AA off makes things better. @olilarkin may have a different opinion, but in general the approach in IGraphics is that things scale and draw nicely and AA seems essential for that, but if there is a really clear case that we haven't taken into account I'd certainly consider it.

**Comment by Youlean** [comment]
@AlexHarker Option to change AA won't mess anything. Disabling AA is present in all 2D graphics engines and it is there for a reason. Not sure why it is such a problem to add this simple feature. 

Anyway, here is the example:

```
    IPattern p1(COLOR_ORANGE);
    //p1.mAntialias = false;

    g.PathRect(IRECT(100, 100, 200.5, 200));
    g.PathFill(p1);

    IPattern p2(COLOR_RED);
    //p2.mAntialias = false;

    g.PathRect(IRECT(200.5, 100, 300, 200));
    g.PathFill(COLOR_RED);
```

![Annotation 2020-05-02 194029](https://user-images.githubusercontent.com/13148293/80871439-ee63c180-8cac-11ea-8ff0-a479a7d7e8b8.png)
![Annotation 2020-05-02 194113](https://user-images.githubusercontent.com/13148293/80871442-f1f74880-8cac-11ea-9a60-83ffbbb8de09.png)
